core/src/main/java/io/questdb/cairo/arr/FlatArrayView.java

```diff
@@ -177,17 +177,21 @@ default int linearSearch(double value, int offset, int length) {
 
     /**
      * Computes the average of the block of elements in this flat array.
+     * Uses Kahan summation.
      *
      * @param offset the starting offset of the block (in elements)
      * @param length the number of elements in the block
      */
     default double sumDouble(int offset, int length) {
-        //TODO: the naive summing algo doesn't compensate for accumulated error
         double sum = 0d;
+        double compensation = 0d;
         for (int i = offset, n = offset + length; i < n; i++) {
             double v = getDoubleAtAbsIndex(i);
-            if (!Double.isNaN(v)) {
-                sum += v;
+            if (!Numbers.isNull(v)) {
```

Comment:
use `Double.isNaN(v)` instead of `Numbers.isNull(v)`:
`Numbers.isNull(v)` treats `infinity` as null, while `Double.isNaN(v)` does not

Reply:
Yes, this is a pre-existing issue in other modules. However, we currently maintain consistency within the array implementation.

Furthermore, the C++ implementation of `FlatArrayView` (`Vect.sumDoubleKahan`) doesn't treat `infinity` as `NaN`:
```
        if (input == input) {
            double y = input - c;
            double t = sum + y;
            c = (t - sum) - y;
            sum = t;
            hasData = true;
        }
```

which is inconsistent with the Java implementation. Therefore, my view is that we should maintain consistency within the array implementation itself, as this brings us closer to the correct behavior.

Reply:
We should follow whatever NumPy would do, I think. It follows IEEE semantics, so an array with an infinity sums to infinity, and an array with both positive and negative infinity sums to NaN.